src/backend/distributed/commands/role.c

```diff
@@ -668,21 +670,42 @@ MakeSetStatementArgument(char *configurationName, char *configurationValue)
 				int intValue;
 				parse_int(configurationValue, &intValue,
 						  (*matchingConfig)->flags, NULL);
-				arg = makeIntConst(intValue, -1);
+				Node *arg = makeIntConst(intValue, -1);
+				args = lappend(args, arg);
 				break;
 			}
 
 			case PGC_REAL:
 			{
-				arg = makeFloatConst(configurationValue, -1);
+				Node *arg = makeFloatConst(configurationValue, -1);
+				args = lappend(args, arg);
 				break;
 			}
 
 			case PGC_BOOL:
 			case PGC_STRING:
 			case PGC_ENUM:
 			{
-				arg = makeStringConst(configurationValue, -1);
+				List *configurationList = NIL;
+
+				if ((*matchingConfig)->flags & GUC_LIST_INPUT)
+				{
+					char *configurationValueCopy = pstrdup(configurationValue);
+					SplitIdentifierString(configurationValueCopy, ',',
+										  &configurationList);
+				}
+				else
+				{
+					configurationList = list_make1(configurationValue);
+				}
+
+				char *configuration = NULL;
+
```

Comment:
nit; this linebreak breaks the logical grouping of `configuration` with the loop it is tied to. Would be great to remove this.

Reply:
done